assert/utils.go

```diff
@@ -491,15 +493,10 @@ func calculateStringSimilarity(target, candidate string) similarItem {
 		return item
 	}
 
-	// 4. Calculate Levenshtein distance
-	distance := levenshteinDistance(target, candidate)
+	// 4. Calculate Damerau-Levenshtein distance for typos
+	distance := damerauLevenshteinDistance(target, candidate)
 	maxLen := max(len(target), len(candidate))
 
-	if maxLen == 0 {
-		item.Similarity = 1.0
-		return item
-	}
-
 	similarity := 1.0 - float64(distance)/float64(maxLen)
```

Comment:
Can't maxLen be 0 and we get a divide my 0 here?

What about this? or something close to this

```suggestion
 	similarity := 1.0
	if maxLen > 0 {
		similarity -= float64(distance)/float64(maxLen)
	}
```